pkg/lib/aws/cloudwatch.go

```diff
@@ -60,7 +64,291 @@ func (c *Client) TagLogGroup(logGroup string, tagMap map[string]string) error {
+	err = json.Unmarshal([]byte(currDashboardString), &currDashboardHolder)
+	if err != nil {
+		return errors.Wrap(err)
+	}
+
+	currDashboard := currDashboardHolder.(map[string]interface{})
+	widgetSlice := currDashboard["widgets"].([]interface{})
+	// append new API metrics widgets to existing widgets
+	widgetSlice = append(widgetSlice, inFlightWidget, latencyWidgetp99, statCodeWidget, apiTitleWidget, latencyWidgetp50)
+
+	currDashboard["widgets"] = widgetSlice
+	currDashboardJSON, err := json.Marshal(currDashboard)
+	if err != nil {
+		return errors.Wrap(err)
+	}
+	// upload updated dashboard to cloudwatch
+	_, err = c.CloudWatch().PutDashboard(&cloudwatch.PutDashboardInput{
+		DashboardName: aws.String(dashboardName),
+		DashboardBody: aws.String(string(currDashboardJSON)),
+	})
+	if err != nil {
+		return errors.Wrap(err)
+	}
+	return nil
+}
+
+// CreateDashboard updates existing dashboard by adding new widgets for new API
+func (c *Client) CreateDashboard(dashboardName string) error {
+
+	// delete Dashboard if already existing exists
+	exists, err := c.DoesDashboardExist(dashboardName)
+	if err != nil {
+		return err
+	}
+	if exists {
+		err = c.DeleteDashboard(dashboardName)
+		if err != nil {
+			return err
+		}
+	}
+
+	//create cloudwatch base body with title
+	cloudwatchBaseBody := map[string]interface{}{"start": "-PT1H", "periodOverride": "inherit", "widgets": []interface{}{createTextWidget(7, 0, 10, 1, "# CORTEX MONITORING DASHBOARD")}}
+	cloudwatchBaseBodyJSON, err := json.Marshal(cloudwatchBaseBody)
+	if err != nil {
+		return errors.Wrap(err, "Failed to encode CLoudwatch Base Body into json")
+	}
+	_, err = c.CloudWatch().PutDashboard(&cloudwatch.PutDashboardInput{
+		DashboardName: aws.String(dashboardName),
+		DashboardBody: aws.String(string(cloudwatchBaseBodyJSON)),
+	})
+	if err != nil {
+		return errors.Wrap(err, "Failed to create Dashboard: "+dashboardName)
 	}
 
 	return nil
 }
+
+// DeleteDashboard deletes dashboard
+func (c *Client) DeleteDashboard(dashboardName string) error {
+
+	var toDelete []*string
+	toDelete = append(toDelete, &dashboardName)
+	_, err := c.CloudWatch().DeleteDashboards(&cloudwatch.DeleteDashboardsInput{
+		DashboardNames: toDelete,
+	})
+	if err != nil {
+		return errors.Wrap(err, "Failed to delete dashboard: "+dashboardName)
+	}
+	return nil
+}
+
+// DoesDashboardExist will check if dashboard with same name as cluster already exists
+func (c *Client) DoesDashboardExist(dashboardName string) (bool, error) {
+	_, err := c.CloudWatch().GetDashboard(&cloudwatch.GetDashboardInput{
+		DashboardName: aws.String(dashboardName),
+	})
+	if err != nil {
+		if CheckErrCode(err, "ResourceNotFound") {
+			return false, nil
+		}
+		return false, errors.Wrap(err, "dashboard: "+dashboardName)
+	}
+
+	return true, nil
+}
+
+// DeleteAPICloudwatch deletes api and reformats cloudwatch
+func (c *Client) DeleteAPICloudwatch(statuses []status.Status, clusterName, apiName string) error {
```

Comment:
Not sure about this. Because `DeleteAPICloudwatch` get called like this
```			
statuses, err := GetAllStatuses()
if err != nil {
	errors.PrintError(err, "failed to get API Statuses")
	return nil
}
// best effort deletion
err = config.AWS.DeleteAPICloudwatch(statuses, config.Cluster.ClusterName, apiName)```
and I would have to create an extra function or loop to create `allAPINames []string`. Do you think this is worth the effort or should I keep it like it is

Reply:
I think it is worth it, since it's only one small extra for loop, and it makes the interface to `DeleteAPICloudwatch()` more clear since all it needs are the API names